### PR TITLE
[4.2] - Admin login - set max width on image

### PR DESCRIPTION
**Pull request by travisrisner**
### Summary of Changes
After updating to 4.2.7, I noticed my custom login images were going outside of the bounds of the box, due to Joomla writing the width in line. I applied a max-width to that image as a possible fix.

### Actual result BEFORE applying this Pull Request
Images larger than the login box go outside of the bounding box.

### Expected result AFTER applying this Pull Request
Images larger than the bounding box on login fit within the box.

### CODE REVIEW

**Comment by ChristineWk** [comment]
The height: auto is absolutely necessary, otherwise the height=xyz from the IMG tag will be used and is then distorted.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39766">issues.joomla.org/tracker/joomla-cms/39766</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :red_circle: unsuccessfully on e6743171a56fc3a2fa71358354fd432189cf5496<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39766">issues.joomla.org/tracker/joomla-cms/39766</a>.</sub>

**Comment by travisrisner** [comment]
> The height: auto is absolutely necessary, otherwise the height=xyz from the IMG tag will be used and is then distorted.
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/39766](https://issues.joomla.org/tracker/joomla-cms/39766).

I'm a bit confused. Are you requesting me change the height to auto? I never said anything about height, and my commit was only setting max width.

**Comment by travisrisner** [comment]
That would be the case if we were setting width, but in this case we are setting max width to ensure images that are too wide for the login box don't go outside the login box. The height style that is set is already in core.

**Comment by drmenzelit** [comment]
max-width solves the problem for images that are too wide for the login module. If your logo is portrait or square and maybe not so wide then the max-width has no effect, but the max-height and the image looks distorted. Normally it should be max-width:100% in combination with height:auto or width: auto and max-height: 4.4rem to allow the image to resize correctly.

**Comment by drmenzelit** [comment]
You can't combine width, max-width and max-height...

Here some screenshots with different CSS definitions (image size 419 x 542px):

max-height: 4.4rem;

grafik

max-width:100%;
height: auto;
grafik

max-height: 4.4rem;
max-width: 100%;
grafik

max-height: 4.4rem;
width: auto;
grafik

**Comment by drmenzelit** [comment]
I think, in the case of the logo for the login module, the height of the image is more important than the width. Imagine if someone puts a 800px high image, the login will looks very weird and you will need to scroll to be able to enter your credentials.
IMHO is max-height: 4.4rem and width: auto the better solution... but that should be tested with different kind of logos.

**Comment by travisrisner** [comment]
Agreed

**Comment by laoneo** [comment]
Not sure if object-fit would help here somehow.

**Comment by drmenzelit** [comment]
> Not sure if object-fit would help here somehow.

I was thinking about that too, but in this case you want to display the full logo, object-fit will crop the image...

**Comment by ChristineWk** [comment]
Took a high picture on purpose. Loaded: 525 x 700 px. :-) But there is no new npm package yet for testing.

pr39766-secondtest

Would above example OK for patch, @drmenzelit?

File: build/media_source/templates/administrator/atum/scss/blocks/_login.scss [hunk 1]
```diff
@@ -38,6 +38,7 @@
 
     img {
       max-height: 4.4rem;
+      width: auto;
```
Comment (Quy) [inline build/media_source/templates/administrator/atum/scss/blocks/_login.scss]:
```suggestion
      width: auto;
      max-height: 4.4rem;
```
> Expected "width" to come before "max-height"  order/properties-order

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7e5315eed803926366c3af5f9180a9233e1618f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39766">issues.joomla.org/tracker/joomla-cms/39766</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 7e5315eed803926366c3af5f9180a9233e1618f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39766">issues.joomla.org/tracker/joomla-cms/39766</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39766">issues.joomla.org/tracker/joomla-cms/39766</a>.</sub>

**Comment by fancyFranci** [comment]
Thank you for fixing that!

**Comment by crystalenka** [comment]
Fix works, but can I suggest a slightly different approach? 4.4em is tiny for a lot of logos.

What about something like:

``` CSS
max-height: 15rem;
max-width: 100%;
object-fit: contain;
```

This bounds the logo by whichever dimension is larger and makes sure it doesn't get distorted, and also makes sure it's not way too tall or too short.

Sorry, I didn't catch this before it was merged!

**Pull request merged**